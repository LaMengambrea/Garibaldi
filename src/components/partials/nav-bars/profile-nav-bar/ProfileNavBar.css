.whiteNavLogo {
    width: 110px;
    float: right;
    margin-top: 15px;
    margin-right: 20px;
}

@media screen and (max-width: 400px) {
    .whiteNavLogo{
        content: url("../../../../content/images/logo_white_clip.png");
        width: 20px;
    }
}

.ProfileNavBar {
    background-color: #4b4b4b;
    color: white;
}

.UserIconMenu {
    float: left;
}

.userFullName {
    float: left;
    padding-top: 16px;
}