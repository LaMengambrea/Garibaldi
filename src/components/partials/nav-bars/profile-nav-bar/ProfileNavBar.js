import React, {Component, PropTypes} from 'react';
import IconMenu from 'material-ui/IconMenu';
import MenuItem from 'material-ui/MenuItem';
import IconButton from 'material-ui/IconButton';
import MoreVertIcon from 'material-ui/svg-icons/navigation/more-vert';
import '../../../../Main.css';
import './ProfileNavBar.css';
import {white} from 'material-ui/styles/colors';

let styles = {
    iconStyle: {
        color: white
    }
}

export default class ProfileNavBar extends Component {
    handleOnItemTouchTap(event, child) {
        console.log("event", event);
        console.log("child", child.props.primaryText);
    }

    render() {
        return (
            <div className="ProfileNavBar row">
                <div className="col-xs-8 col-md-6">
                    <IconMenu
                        iconButtonElement={<IconButton><MoreVertIcon /></IconButton>}
                        anchorOrigin={{horizontal: 'left', vertical: 'top'}}
                        targetOrigin={{horizontal: 'left', vertical: 'top'}}
                        animated={true}
                        onItemTouchTap={this.handleOnItemTouchTap}
                        iconStyle={styles.iconStyle}
                        className={"UserIconMenu"}
                        >
                        <MenuItem primaryText="Mi perfil" />
                        <MenuItem primaryText="Cerrar sesiÃ³n" />
                    </IconMenu>
                    <div className="userFullName">{`${this.props.user.name} ${this.props.user.lastName}`}</div>
                </div>
                <div className="col-xs-4 col-md-6">
                    <img src="" alt="" className="whiteNavLogo"/>
                </div>
            </div>
        );
    }
}

ProfileNavBar.displayName = 'ProfileNavBar'

ProfileNavBar.propTypes = {
  user: PropTypes.object
};