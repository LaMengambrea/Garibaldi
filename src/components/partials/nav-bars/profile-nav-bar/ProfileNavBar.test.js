import ProfileNavBar from './ProfileNavBar';
import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

const user = {
    username: "username",
    password: "password",
    email: "email@gmail.com",
    name: "name",
    lastName: "lastName",
    birthDate: "birthDate",
    address1: "Acordada #23",
    address2: "Lomas Verdes",
    city: "México",
    state: "Estado de México",
    country: "México",
    userType: "1"
}

let store = {
    subscribe: () => {},
    dispatch: () => {},
    getState: () => {
      return {
          receiveCurrentUser: jest.fn()
      };
    }
}

it('ProfileNavBar renders', () => {
  const tree = renderer.create(
    <Provider store={store}>
        <MuiThemeProvider>
            <ProfileNavBar user={user}/>
        </MuiThemeProvider>
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});