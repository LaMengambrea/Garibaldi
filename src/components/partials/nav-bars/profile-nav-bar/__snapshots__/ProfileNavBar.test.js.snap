exports[`test ProfileNavBar renders 1`] = `
<div
  className="ProfileNavBar row">
  <div
    className="col-xs-8 col-md-6">
    <div
      className="UserIconMenu"
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchTap={[Function]}
      style={
        Object {
          "display": "inline-block",
          "muiPrepared": true,
          "position": "relative",
        }
      }>
      <button
        disabled={false}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOut={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "background": "none",
            "border": 10,
            "boxSizing": "border-box",
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 0,
            "fontWeight": "inherit",
            "height": 48,
            "margin": 0,
            "muiPrepared": true,
            "outline": "none",
            "overflow": "visible",
            "padding": 12,
            "position": "relative",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "verticalAlign": null,
            "width": 48,
            "zIndex": 1,
          }
        }
        tabIndex={0}
        type="button">
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}>
          <svg
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "color": "#ffffff",
                "display": "inline-block",
                "fill": "currentColor",
                "height": 24,
                "muiPrepared": true,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "width": 24,
              }
            }
            viewBox="0 0 24 24">
            <path
              d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z" />
          </svg>
        </div>
      </button>
      <div
        style={
          Object {
            "display": "none",
          }
        } />
    </div>
    <div
      className="userFullName">
      name lastName
    </div>
  </div>
  <div
    className="col-xs-4 col-md-6">
    <img
      alt=""
      className="whiteNavLogo"
      src="" />
  </div>
</div>
`;
